conda/shell/Scripts/activate.bat

```diff
@@ -1,28 +1,41 @@
-@REM Copyright (C) 2012 Anaconda, Inc
-@REM SPDX-License-Identifier: BSD-3-Clause
-@REM Test first character and last character of %1 to see if first character is a "
-@REM   but the last character isn't.
-@REM This was a bug as described in https://github.com/ContinuumIO/menuinst/issues/60
-@REM When Anaconda Prompt has the form
-@REM   %windir%\system32\cmd.exe "/K" "C:\Users\builder\Miniconda3\Scripts\activate.bat" "C:\Users\builder\Miniconda3"
-@REM Rather than the correct
-@REM    %windir%\system32\cmd.exe /K ""C:\Users\builder\Miniconda3\Scripts\activate.bat" "C:\Users\builder\Miniconda3""
-@REM this solution taken from https://stackoverflow.com/a/31359867
-@set "_args1=%1"
-@set _args1_first=%_args1:~0,1%
-@set _args1_last=%_args1:~-1%
-@set _args1_first=%_args1_first:"=+%
-@set _args1_last=%_args1_last:"=+%
-@set _args1=
-
-@if "%_args1_first%"=="+" if NOT "%_args1_last%"=="+" (
-    @CALL "%~dp0..\condabin\conda.bat" activate
-    @GOTO :End
-)
-
-@REM This may work if there are spaces in anything in %*
-@CALL "%~dp0..\condabin\conda.bat" activate %*
-
-:End
-@set _args1_first=
-@set _args1_last=
+:: Copyright (C) 2012 Anaconda, Inc
+:: SPDX-License-Identifier: BSD-3-Clause
+
+:: disable displaying the command before execution
+@ECHO OFF
+
+:: skip checking for quotes if there are no arguments
+IF [%1]==[] GOTO :SKIP_CHECK
+
+:: enter localized variable scope
+SETLOCAL
+
+:: Test first character and last character of %1 to see if first character is a "
+::   but the last character isn't.
+:: This was a bug as described in https://github.com/ContinuumIO/menuinst/issues/60
+:: When Anaconda Prompt has the form
+::   %windir%\system32\cmd.exe "/K" "C:\Users\builder\Miniconda3\Scripts\activate.bat" "C:\Users\builder\Miniconda3"
+:: Rather than the correct
+::    %windir%\system32\cmd.exe /K ""C:\Users\builder\Miniconda3\Scripts\activate.bat" "C:\Users\builder\Miniconda3""
+:: this solution taken from https://stackoverflow.com/a/31359867
+
+SET "_ARG=%1"
+SET _ARG_FIRST=%_ARG:~0,1%
+SET _ARG_LAST=%_ARG:~-1%
+SET _ARG_FIRST=%_ARG_FIRST:"=+%
+SET _ARG_LAST=%_ARG_LAST:"=+%
+
+:: if the first character is not a quote we can skip further quote checking
+IF NOT [%_ARG_FIRST%]==[+] ENDLOCAL & GOTO :SKIP_CHECK
+
+:: the first and last characters appear to be matching quotes so we can activate normally
+IF [%_ARG_LAST%]==[+] ENDLOCAL & GOTO :SKIP_CHECK
+
+:: found a quote mismatch, we assume this is a bug in the downstream code
+ENDLOCAL & CALL "%~dp0\..\condabin\conda.bat" activate
+GOTO :EOF
+
+:SKIP_CHECK
```

Comment:
Sorry, I missed the tag. Right now, the target of the shortcut looks like this

```
%windir%\System32\cmd.exe "/K" C:\Users\<user>\miniconda3\Scripts\activate.bat C:\Users\<user>\miniconda3
```

So, maybe install `console_shortcut_miniconda` or `console_shortcut` with the dev version of `conda` and see if the workaround is still necessary?

if it is, removing it is tricky because that would render that shortcut unusable when a user updates `conda`.